### PR TITLE
Contact force to auxkernel 4850

### PR DESCRIPTION
**Pull request by shartibartfast**
This reverts the previous implementation storing/loading PenetrationInfo:_contact_force from aux variables inside of MechanicalContactConstraint and moves those tasks to an AuxKernel.

I left the reverts unsquashed to make it easier to see what has changed.

### CODE REVIEW

**Comment by shanestafford** [comment]
I ran some assessment cases with this and it appears to work fine now. From my perspective, it is ready to go.

**Comment by jwpeterson** [comment]
Is this the implementation @bwspenc did not like as much?  Presumably he changed his mind?

**Comment by shanestafford** [comment]
That was back when we thought the other method would work. The old method fails in parallel on random cases because there is no clean/easy way to sync writes to the aux system at the end of a time step from within a Constraint. But I'll let Ben speak for himself.

File: modules/contact/src/base/ContactApp.C [hunk 1]
```diff
@@ -90,7 +93,8 @@ ContactApp::associateSyntax(Syntax & syntax, ActionFactory & action_factory)
   syntax.registerActionSyntax("NodalAreaAction", "Contact/*");
   syntax.registerActionSyntax("NodalAreaVarAction", "Contact/*");
 
-  registerAction(ContactAction, "add_aux_variable");
+  syntax.registerActionSyntax("PenetrationInfoAuxAction", "Contact/*");
+
   registerAction(ContactAction, "add_dg_kernel");
```
Comment (bwspenc) [inline modules/contact/src/base/ContactApp.C]: ContactAction doesn't have an ```add_dg_kernel``` action anymore, right?  This should be removed.

File: modules/contact/src/PenetrationInfoAuxAction.C [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/****************************************************************/
```
Comment (bwspenc) [inline modules/contact/src/PenetrationInfoAuxAction.C]: Why not just do this in the ContactAction along with everything else?  I guess it's just a personal taste thing, but I think it's confusing to have a bunch of separate Action classes to set up different things.

**Comment by bwspenc** [comment]
Yes, I've been convinced that this is the way to do it.

File: modules/contact/src/PenetrationInfoAuxAction.C (hunk 2 above)
Comment (shanestafford) [inline modules/contact/src/PenetrationInfoAuxAction.C]: I'll make this change.

File: modules/contact/src/base/ContactApp.C (hunk 1 above)
Comment (shanestafford) [inline modules/contact/src/base/ContactApp.C]: I'll make this change too.

**Comment by shanestafford** [comment]
@bwspenc is this more like what you want?

**Comment by bwspenc** [comment]
Looks good.

**Comment by shanestafford** [comment]
Still not working on the assessment cases!

**Comment by shanestafford** [comment]
Looks like the assessment cases are working again. I think this PR is ready to go.

**Pull request merged**